samtranslator/model/api/api_generator.py

```diff
@@ -269,6 +276,61 @@ def _add_auth(self):
         # Assign the Swagger back to template
         self.definition_body = swagger_editor.swagger
 
+    def _add_gateway_responses(self):
+        """
+        Add Gateway Response configuration to the Swagger file, if necessary
+        """
+
+        if not self.gateway_responses:
+            return
+
+        if self.gateway_responses and not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id, "GatewayResponses works only with inline Swagger specified in "
+                                 "'DefinitionBody' property")
+
+        # Make sure keys in the dict are recognized
+        for responses_key, responses_value in self.gateway_responses.items():
```

Comment:
I've moved most of the validation there. I left behind the validation for the presence of properties other than `ResponseParameters`, `ResponseTemplates` and `StatusCode` because they're accessed in order to _create_ an `ApiGatewayResponse`.

But if ignored, overflow properties are actually OK, I can scratch it entirely.